src/components/dialogs/DialogDelegations.tsx

```diff
@@ -0,0 +1,103 @@
+import { useDialog } from '@/stores'
+import {
+  Box,
+  Button,
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormControlLabel,
+  Typography,
+} from '@mui/material'
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { useNavigate } from '@/router'
+import { DelegationMutations } from '@/api/delegation'
+import { EServiceMutations } from '@/api/eservice'
+import { EServiceCreateDraftValues } from '@/pages/DelegationCreatePage/components/DelegationCreateForm'
+
+/**
+ * eserviceParams are used when creating a new eservice;
+ * delegationParams has eserviceId optional because it is passed only when the eservice already exists
+ */
+export type DialogDelegationsProps = {
+  eserviceParams?: EServiceCreateDraftValues
+  delegationParams: { eserviceId?: string; delegateId: string }
+}
+
+export const DialogDelegations: React.FC<DialogDelegationsProps> = ({
+  eserviceParams,
+  delegationParams,
+}) => {
+  const ariaLabelId = React.useId()
+
+  const { t } = useTranslation('party', {
+    keyPrefix: 'delegations.create.dialog',
+  })
+
+  const { closeDialog } = useDialog()
+
+  const [isCheckboxChecked, setIsCheckboxChecked] = React.useState<boolean>(false)
+
+  const { mutate: createProducerDelegation } = DelegationMutations.useCreateProducerDelegation()
+  const { mutate: createProducerEserviceDraft } = EServiceMutations.useCreateDraft()
+
+  const handleCheckBoxChange = () => {
+    setIsCheckboxChecked((prev) => {
+      return !prev
+    })
+  }
+
+  const navigate = useNavigate()
+
+  const onSubmit = () => {
+    let createDelegationParams = {
+      eserviceId: delegationParams?.eserviceId || '',
+      delegateId: delegationParams?.delegateId,
+    }
+    if (eserviceParams) {
+      createProducerEserviceDraft(eserviceParams, {
+        onSuccess({ id }) {
+          createDelegationParams = {
+            eserviceId: id,
+            delegateId: delegationParams?.delegateId,
+          }
+          createProducerDelegation(createDelegationParams)
+        },
+      })
+    }
+    createProducerDelegation(createDelegationParams)
+    navigate('DELEGATIONS')
+    closeDialog()
+  }
```

Comment:
Let's not put the submit logic here, maybe the dialog can have a `onConfirm` prop?

Plus the logic as is would not work.
First, the `eserviceParams` are truthy the `createProducerDelegation` would be called twice.

There must be a polling system inside the `onSuccess` since the e-service creation does not happen synchrounously. 

There is no need to have `createDelegationParams` as a `let` and the reassignment at line 62